src/page/couponCreation/CouponCreation.js

```diff
@@ -0,0 +1,60 @@
+import OptionList from "../../component/optionList/OptionList";
+import * as couponOptions from "../../data/couponCreation";
+import * as ccs from "./couponCreationStyle";
+import {useCallback, useState} from "react";
+import {generateRandomString} from "../../data/dateGenerator";
+
+function CouponCreation() {
+
+  const [selectedCouponOptions, setSelectedCouponOptions] = useState({
+    coach: '', couponType: '', coupons: 0, couponNumbers: []
+  });
+
+  const handleChange = useCallback((optionName, {target: {value = ''}}) => {
+    setSelectedCouponOptions({
+      ...selectedCouponOptions, [optionName]: value
+    })
```

Comment:
```suggestion
    setSelectedCouponOptions(prevSelectedCouponOptions => {
      prevSelectedCouponOptions, [optionName]: value
    })
```
이거는 지금 당장은 문제가 없는데요.
리액트에서 setState는 비동기로 처리됩니다. 
state의 지난 값을 참조하여 새로운 state를 만드는 경우 state를 직접 참조하여 값을 가져오는 것 보다 setState 내부에서 받아오는게 가장 정확합니다.

[setState 비동기](https://ko.reactjs.org/docs/faq-state.html#why-is-setstate-giving-me-the-wrong-value)

Reply:
setState((prev)=>{/*code*/}) 좋아용

Reply:
이 내용을 공부했던 기억이 있는대 까먹고 있었네요.. 피드백 좋아요😁